### DISCUSSION TITLE
Best way of resolving workspace dependencies in a monorepo

### DISCUSSION THREAD

**Discussion by Llois41** [vitest-dev/vitest#5517] [Q&A]
We have a monorepo which looks (a little bit simplified) like this:

```
\repo
    \package1
        \src
    \package2
    \package3
```

Imports of those packages look like `import { someFunc } from '@repo/package1'`.
In the package.json it is referred to as `"@repo/package1": "workspace:*"` (since we use yarn workspaces).

Using Jest this seems to resolve out-of-the-box. Switching to vitest throws 

```
someFunc is not a function
TypeError: __vite_ssr_import_3__.someFunc is not a function
```

I can resolve this by using

```
resolve: {
    alias: { '@repo/package1': path.join(__dirname, '..', 'package1/') },
  },
```

Is there a way to not add all of my packages to the alias list (e.g. by using a regex)?

**Comment by hi-ogawa** [comment]
Can you provide a reproduction? Probably `import { someFunc } from '@repo/package1'` should just work without alias.

I don't think Vitest has a bug around this area, but potentially something confusing is happening due to packaging issues or yarn workspaces's `node_modules`, so having a reproduction would help us identifying potential mis-configurations.

If you want to go with `resolve.alias`, then you can probably try `find/replace` https://github.com/rollup/plugins/tree/master/packages/alias#arrayobject-format. I don't remember the exact syntax, but maybe something like this:

```ts
    alias: [
      {
        find: /^@repo\/(.*)$/,
        replacement: path.join(__dirname, '..', '$1'),
      }
    ]
```
